File: app/src/main/java/com/mahmoud/mostafa/marvel/data/api/ApiInterface.java
package com.mahmoud.mostafa.marvel.data.api;

import com.mahmoud.mostafa.marvel.data.pojos.characters.Character;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiInterface {
    @GET("characters")
    Call<Character> getCharacters(@Query("ts") String timeStamp, @Query("apikey") String apiKey, @Query("hash") String hash);


    @GET("characters")
    Call<Character> getCharactersByName(@Query("ts") String timeStamp, @Query("apikey") String apiKey, @Query("hash")
            String hash, @Query("nameStartsWith") String name);

    @GET("characters/{characterId}/comics")
    Call<Character> getComics(@Path("characterId") String characterId, @Query("ts") String timeStamp,
                                    @Query("apikey") String apiKey, @Query("hash") String hash);


    @GET("characters/{characterId}/events")
    Call<Character> getEvents(@Path("characterId") String characterId, @Query("ts") String timeStamp,
                                    @Query("apikey") String apiKey, @Query("hash") String hash);

    @GET("characters/{characterId}/series")
    Call<Character> getSeries(@Path("characterId") String characterId, @Query("ts") String timeStamp,
                                    @Query("apikey") String apiKey, @Query("hash") String hash);


    @GET("characters/{characterId}/stories")
    Call<Character> getStories(@Path("characterId") String characterId, @Query("ts") String timeStamp,
                                     @Query("apikey") String apiKey, @Query("hash") String hash);


}

File: app/src/main/java/com/mahmoud/mostafa/marvel/data/api/ApiInterface.kt
package com.mahmoud.mostafa.marvel.data.api

import com.mahmoud.mostafa.marvel.data.pojos.characters.Character

import io.reactivex.Observable
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiInterface {
    @GET("characters")
    fun getCharacters(@Query("ts") timeStamp: String, @Query("apikey") apiKey: String, @Query("hash") hash: String): Call<Character>


    @GET("characters")
    fun getCharactersByName(@Query("ts") timeStamp: String, @Query("apikey") apiKey: String, @Query("hash")
    hash: String, @Query("nameStartsWith") name: String): Call<Character>

    @GET("characters/{characterId}/comics")
    fun getComics(@Path("characterId") characterId: String, @Query("ts") timeStamp: String,
                  @Query("apikey") apiKey: String, @Query("hash") hash: String): Call<Character>


    @GET("characters/{characterId}/events")
    fun getEvents(@Path("characterId") characterId: String, @Query("ts") timeStamp: String,
                  @Query("apikey") apiKey: String, @Query("hash") hash: String): Call<Character>

    @GET("characters/{characterId}/series")
    fun getSeries(@Path("characterId") characterId: String, @Query("ts") timeStamp: String,
                  @Query("apikey") apiKey: String, @Query("hash") hash: String): Call<Character>


    @GET("characters/{characterId}/stories")
    fun getStories(@Path("characterId") characterId: String, @Query("ts") timeStamp: String,
                   @Query("apikey") apiKey: String, @Query("hash") hash: String): Call<Character>


}
